x-pack/plugins/fleet/server/routes/package_policy/handlers.ts

```diff
@@ -263,6 +265,10 @@ export const createPackagePolicyHandler: FleetRequestHandler<
       } as NewPackagePolicy);
     }
 
+    if (newPackagePolicy.inputs && newPackagePolicy.inputs.length) {
+      validateEndpointPackagePolicy(newPackagePolicy.inputs[0]);
```

Comment:
I see this being the only package specific action done here. 
Why we don't do this at our `packagePolicyCreate` external callback as we do for other Endpoint specific validations?

Reply:
Example of external callback call at service level: https://github.com/elastic/kibana/blob/main/x-pack/plugins/fleet/server/services/package_policy.ts#L174

Example of external callback usage and implementation: https://github.com/elastic/kibana/blob/main/x-pack/plugins/security_solution/server/fleet_integration/fleet_integration.ts#L77

Where we already do some validations.

Reply:
The reason I went with handler and not callbacks is that at this point I believe that error handling of these callbacks is broken. Throwing error with attached `statusCode` (see https://github.com/elastic/kibana/blob/6efef0496077f4e61d49e4e43f75941ff3d98d9e/x-pack/plugins/security_solution/server/fleet_integration/handlers/validate_policy_against_license.ts#L13) doesn't result in fleet api returning said response and error message. Try changing `if` I posted above to `if (true)` so it always fires, error can be seen in the console but we end up with `200` response. This happens, I believe, because over [here](https://github.com/elastic/kibana/blob/56be6c6fb6f535c250cde9859310c3ba2337cefe/x-pack/plugins/fleet/server/services/package_policy.ts#L697-L709) catched error doesn't have any effect on the response but the cli error.

Reply:
Not sure if I understood it correctly. If I throw an error in [getPackagePolicyCreateCallback](https://github.com/elastic/kibana/blob/main/x-pack/plugins/security_solution/server/fleet_integration/fleet_integration.ts#L77) I see API returning 403 with the thrown error in it.

Seems you are correct on the example above for the Update one. Then I wonder if we can track the errors at `catch` level and:
- Fix the existing issue where the error is not thrown as an API failure.
- Catch the new validation error.

Let me know what do you think.